### DISCUSSION TITLE
Overriding detected file kind

### DISCUSSION THREAD

**Discussion by rokcarl** [ansible/ansible-lint#1501] [General]
I've upgraded ansible-lint from 4.0.0 to 5.0.6. When running it on one of our yaml files, I get this:

```bash
$ ansible-lint -p ./playbooks/pipeline/site.yml
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: ./playbooks/pipeline/site.yml
...
```

Why do I get this warning? I've googled, found #1368 and #1334, but wasn't really explained. Found it in code [here](https://github.com/ansible-community/ansible-lint/blob/878a90edbce2a985837d26883bff636ec9c7c3a7/src/ansiblelint/utils.py#L784), but still didn't make much sense to me. Why am I getting the warning and how to fix it?

Also, why do I get lint errors from multiple files, if I only pass it one file? Does it parse this file and then lint all the files that depend on it as well?

**Comment by anselal** [comment]
> I've upgraded ansible-lint from 4.0.0 to 5.0.6. When running it on one of our yaml files, I get this:
> 
> ```shell
> $ ansible-lint -p ./playbooks/pipeline/site.yml
> WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: ./playbooks/pipeline/site.yml
> ...
> ```
> 
> Why do I get this warning? I've googled, found #1368 and #1334, but wasn't really explained. Found it in code [here](https://github.com/ansible-community/ansible-lint/blob/878a90edbce2a985837d26883bff636ec9c7c3a7/src/ansiblelint/utils.py#L784), but still didn't make much sense to me. Why am I getting the warning and how to fix it?
> 
> Also, why do I get lint errors from multiple files, if I only pass it one file? Does it parse this file and then lint all the files that depend on it as well?

https://github.com/ansible/ansible-lint/discussions/1679?sort=new
